azure-toolkit-libs/azure-toolkit-appservice-lib/src/main/java/com/microsoft/azure/toolkits/appservice/model/Runtime.java

```diff
@@ -34,8 +33,49 @@
     public static final Runtime LINUX_JAVA11_TOMCAT85 = new Runtime(OperatingSystem.LINUX, WebContainer.TOMCAT_85, JavaVersion.JAVA_11);
     public static final Runtime DOCKER = new Runtime(OperatingSystem.DOCKER, null, null);
 
+    private static final List<Runtime> values = Collections.unmodifiableList(Arrays.asList(WINDOWS_JAVA8, WINDOWS_JAVA11, WINDOWS_JAVA8_TOMCAT9,
+            WINDOWS_JAVA8_TOMCAT85, WINDOWS_JAVA11_TOMCAT9, WINDOWS_JAVA11_TOMCAT85, LINUX_JAVA8, LINUX_JAVA11, LINUX_JAVA8_TOMCAT9, LINUX_JAVA8_TOMCAT85,
+            LINUX_JAVA8_JBOSS72, LINUX_JAVA11_TOMCAT9, LINUX_JAVA11_TOMCAT85));
+
+    private OperatingSystem operatingSystem;
+    private WebContainer webContainer;
+    private JavaVersion javaVersion;
+
+    public static Runtime getRuntime(OperatingSystem operatingSystem, WebContainer webContainer, JavaVersion javaVersion) {
+        return values().stream()
+                .filter(runtime -> Objects.equals(runtime.operatingSystem, operatingSystem))
+                .filter(runtime -> Objects.equals(runtime.webContainer, webContainer))
+                .filter(runtime -> Objects.equals(runtime.javaVersion, javaVersion))
+                .findFirst().orElse(new Runtime(operatingSystem, webContainer, javaVersion));
```

Comment:
return a non-existing one? or throw a runtime exception?

Reply:
For Runtime, we just list common runtime here, still support users use customized java version (like 1.8.0_*)

Reply:
add a warning for non-standard runtime